doc/releases/changelog-0.36.0.md

```diff
@@ -303,68 +383,90 @@
   [(#5539)](https://github.com/PennyLaneAI/pennylane/pull/5539)
   
-* `ApproxTimeEvolution` is now compatible with any operator that defines a `pauli_rep`.
+* `qml.ApproxTimeEvolution` is now compatible with any operator that has a defined `pauli_rep`.
   [(#5362)](https://github.com/PennyLaneAI/pennylane/pull/5362)
 
-* `Hamiltonian.pauli_rep` is now defined if the hamiltonian is a linear combination of paulis.
+* `Hamiltonian.pauli_rep` is now defined if the hamiltonian is a linear combination of Pauli operators.
   [(#5377)](https://github.com/PennyLaneAI/pennylane/pull/5377)
 
-* `Prod.eigvals()` is now compatible with Qudit operators.
+* `Prod` instances created with qutrit operators now have a defined `eigvals()` method.
   [(#5400)](https://github.com/PennyLaneAI/pennylane/pull/5400)
 
-* `qml.transforms.hamiltonian_expand` can now handle multi-term observables with a constant offset.
+* `qml.transforms.hamiltonian_expand` can now handle multi-term observables with a constant offset (i.e., terms like `qml.I()`).
   [(#5414)](https://github.com/PennyLaneAI/pennylane/pull/5414)
 
-* `taper_operation` method is compatible with new operator arithmetic.
+* `qml.qchem.taper_operation` is now compatible with the new operator arithmetic.
   [(#5326)](https://github.com/PennyLaneAI/pennylane/pull/5326)
 
-* Removed the warning that an observable might not be hermitian in `qnode` executions. This enables jit-compilation.
+* The warning for an observable that might not be hermitian in QNode executions has been removed. This enables jit-compilation.
   [(#5506)](https://github.com/PennyLaneAI/pennylane/pull/5506)
 
 * `qml.transforms.split_non_commuting` will now work with single-term operator arithmetic.
   [(#5314)](https://github.com/PennyLaneAI/pennylane/pull/5314)
 
-* `LinearCombination` and `Sum` now accept `_grouping_indices` on initialization.
+* `LinearCombination` and `Sum` now accept `_grouping_indices` on initialization. This addition is relevant to developers only. 
   [(#5524)](https://github.com/PennyLaneAI/pennylane/pull/5524)
 
+* Calculating the dense, differentiable matrix for `PauliSentence` and operators with Pauli sentences
+  is now faster.
+  [(#5578)](https://github.com/PennyLaneAI/pennylane/pull/5578)
+
+<h4>Community contributions 🥳</h4>
+
+* `ExpectationMP`, `VarianceMP`, `CountsMP`, and `SampleMP` now have a 
+  `process_counts` method (similar to `process_samples`). This allows for 
+  calculating measurements given a `counts` dictionary.
+  [(#5256)](https://github.com/PennyLaneAI/pennylane/pull/5256)
+  [(#5395)](https://github.com/PennyLaneAI/pennylane/pull/5395)
+
+* Type-hinting has been added in the `Operator` class for better interpretability.
+  [(#5490)](https://github.com/PennyLaneAI/pennylane/pull/5490)
+
+* An alternate strategy for sampling with multiple different `shots` values has 
+  been implemented via the `shots.bins()` method.
+  [(#5476)](https://github.com/PennyLaneAI/pennylane/pull/5476)
```

Comment:
These changelog entries weren't clear so I changed them quite a bit. Would be good to have a second look here to make sure I understood the impact of the features and conveyed it properly :)